src/main/java/org/jenkinsci/maven/plugins/hpi/TestDependencyMojo.java

```diff
@@ -583,28 +592,30 @@ private static void applyOverrides(
             }
         }
 
-        /*
-         * If a bundled plugin was added that is neither in the model nor the transitive dependency
-         * chain, add a test-scoped direct dependency to the model. This is necessary in order for
-         * us to be able to correctly populate target/test-dependencies/ later on.
-         */
-        Set<String> unappliedBundledPlugins = new HashSet<>(bundledPlugins.keySet());
-        unappliedBundledPlugins.removeAll(appliedBundledPlugins);
-        for (String key : unappliedBundledPlugins) {
-            String[] groupArt = key.split(":");
-            String groupId = groupArt[0];
-            String artifactId = groupArt[1];
-            String version = bundledPlugins.get(key);
-            Dependency dependency = new Dependency();
-            dependency.setGroupId(groupId);
-            dependency.setArtifactId(artifactId);
-            dependency.setVersion(version);
-            dependency.setScope(Artifact.SCOPE_TEST);
-            if (dependency.getGroupId().equals(project.getGroupId()) && dependency.getArtifactId().equals(project.getArtifactId())) {
-                throw new MojoExecutionException("Cannot add self as test-scoped dependency");
-            }
-            log.info(String.format("Adding test-scoped direct dependency %s:%s", key, version));
-            project.getDependencies().add(dependency);
+        if (overrideWarAdditions) {
+            /*
+            * If a bundled plugin was added that is neither in the model nor the transitive dependency
+            * chain, add a test-scoped direct dependency to the model. This is necessary in order for
+            * us to be able to correctly populate target/test-dependencies/ later on.
+            */
```

Comment:
These stars and bars comments are not indented properly:

```suggestion
            /*
             * If a bundled plugin was added that is neither in the model nor the transitive dependency
             * chain, add a test-scoped direct dependency to the model. This is necessary in order for
             * us to be able to correctly populate target/test-dependencies/ later on.
             */
```